### DISCUSSION TITLE
RenderBinDetails and annotation groups...

### DISCUSSION THREAD

**Discussion by keince** [pelicanmapping/osgearth#1930] [Support Q&A]
I have two groups where each group has annotation type TrackNode for the nodes in each group. I need to have one group to always render on top of the other group.

For group A, I set 

groupA->getOrCreateStateSet()->setRenderBinDetails(3, "RenderBin");
mapNode->addChild(groupA);

For group B, I set

groupB->getOrCreateStateSet()->setRenderBinDetails(INT_MAX, "RenderBin");
mapNode->addChild(groupB);

so that group B is always rendered on top of group A.

This seems to have no effect. No matter what I do, group A is always on top of group B.

Is there any way I can get group B to render on top of group A?

Thanks

Shayne

**Comment by chenguanga** [comment]
You can try to close depth test:
groupA->getOrCreateStateSet()->setMode(GL_DEPTH_TEST,osg::StateAttribute::OFF);

**Reply by keince** [reply]
I appreciate the feedback. 

It appears for TrackNodes that the depth buffer is completely disabled so I'm not sure turning the depth test off for groupA will resolve my issue.

It looks like all annotations (including TrackNode) are set to always use renderbin 13. I think I will need to figure out a way to set the TrackNode groupB to use something like renderbin 14 to get them to always be on top.
